compile.c

```diff
@@ -13270,7 +13269,6 @@ rb_iseq_ibf_dump(const rb_iseq_t *iseq, VALUE opt)
 
     str = dump->global_buffer.str;
     ibf_dump_free(dump);
```

Comment:
Since we are no longer setting the pointer to `NULL`, we will have a double free here since the GC will free everything again.

```suggestion
```

Reply:
Actually, the code handles that:
```c
static void
ibf_dump_free(void *ptr)
{
    struct ibf_dump *dump = (struct ibf_dump *)ptr;
    if (dump->global_buffer.obj_table) {
        st_free_table(dump->global_buffer.obj_table);
        dump->global_buffer.obj_table = 0;
    }
    if (dump->iseq_table) {
        st_free_table(dump->iseq_table);
        dump->iseq_table = 0;
    }
}
```

Reply:
I'm not sure if it's really worth keeping that early free thought.

Reply:
IMO it's a better idea to leave freeing to the GC.

Reply:
Alright, I suspected it was added for perf reasons, hence why I was hesistant, but after blaming it appears it's a "left-over": https://github.com/ruby/ruby/commit/2c3f0749c56a1a094ca7be12e669f0d1545bb756

I'll remove it.